### PR TITLE
DROOLS-2569 [DMN Designer] Marshalling of magnet positions

### PR DESCRIPTION
**Pull request by tarilabs**
Association DMN Edge DMNDI serialization.

/cc @manstis @jomarko this PR should address this last issue: https://issues.jboss.org/browse/DROOLS-2569?focusedCommentId=13654081&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13654081

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 1]
```diff
@@ -1358,6 +1364,64 @@ public void testStunnerConstellationButtonCausingPoint2DbeingNull() throws IOExc
...
+    @Test
+    public void testAssociationEdgeDMNDI() throws IOException {
+        Diagram diagram = applicationFactoryManager.newDiagram("testDiagram",
+                                                               BindableAdapterUtils.getDefinitionSetId(DMNDefinitionSet.class),
+                                                               null);
+        Graph g = diagram.getGraph();
+        Node diagramRoot = DMNMarshaller.findDMNDiagramRoot(g);
+        testAugmentWithNSPrefixes(((DMNDiagram) ((View<?>) diagramRoot.getContent()).getDefinition()).getDefinitions());
+
+        org.kie.dmn.model.api.InputData dmnInputData = (org.kie.dmn.model.api.InputData) new TInputData();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: Casting is not needed as `TInputData` implements `InputData`.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 2]
```diff
@@ -1358,6 +1364,64 @@ public void testStunnerConstellationButtonCausingPoint2DbeingNull() throws IOExc
...
+        dmnInputData.setId("inputDataID");
+        dmnInputData.setName(dmnInputData.getId());
+        Node inputDataNode = new InputDataConverter(applicationFactoryManager).nodeFromDMN(dmnInputData);
+        org.kie.dmn.model.api.TextAnnotation dmnTextAnnotation = (org.kie.dmn.model.api.TextAnnotation) new TTextAnnotation();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: Casting is not needed as `TTextAnnotation` implements `TextAnnotation`.

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 3]
```diff
@@ -1358,6 +1364,64 @@ public void testStunnerConstellationButtonCausingPoint2DbeingNull() throws IOExc
...
+        dmnTextAnnotation.setId("decisionID");
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: I am not sure about "decisionID" as we are connecting InputData and TextAnnotation in the test. Could we use more appropriate name please?

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 4]
```diff
@@ -1358,6 +1364,64 @@ public void testStunnerConstellationButtonCausingPoint2DbeingNull() throws IOExc
...
+        DMNMarshaller.connectRootWithChild(diagramRoot, inputDataNode);
+        DMNMarshaller.connectRootWithChild(diagramRoot, textAnnotationNode);
+
+        DMNMarshaller m = new DMNMarshaller(new XMLEncoderDiagramMetadataMarshaller(),
+                                            applicationFactoryManager);
+        String output = m.marshall(diagram);
+        LOG.debug(output);
+
+        Definitions dmnDefinitions = DMNMarshallerFactory.newDefaultMarshaller().unmarshal(output);
+        assertThat(dmnDefinitions.getDMNDI().getDMNDiagram().get(0).getDMNDiagramElement().stream().filter(DMNEdge.class::isInstance).count()).isEqualTo(1);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: We can shorten this line slightly:
```java
assertThat(dmnDefinitions.getDMNDI().getDMNDiagram().get(0).getDMNDiagramElement().stream().filter(DMNEdge.class::isInstance)).hasSize(1);
```

Comment (tarilabs) [reply kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: will keep on rely on JDK method as preference.

**Comment by tarilabs** [comment]
Can this be merged, please.

**Comment by manstis** [comment]
@tarilabs For sure.. just need a nudge sometimes when we've been waiting for the PR to turn green.

**Pull request merged**